ssp

```diff
@@ -8826,6 +8828,22 @@ sub check_for_cpro {
 
 ## END malware checks
 
+sub detect_slowloris {
+    my $sockcount = 75;
+    my $count = 0;
+
+    return unless my $netstat_out = timed_run( 0, 'netstat', '-nap' );
+    for my $line ( split( '\n', $netstat_out ) ) {
+        if ( $line =~ m/ESTABLISHED.*httpd/ ) {
+            $count++;
+        }
+    }
+    if ( $count >= $sockcount ) {
+        print_warn("Possible SLOWLORIS attack: Found ".$count." Apache sockets in ESTABLISHED\n");
+        print_warning("\t \\_ See https://documentation.cpanel.net/display/EA/How+To+Mitigate+Slowloris+Attacks");
+    }
+}
+
```

Comment:
Typically, yes, they are a single IP address. I have, however, seen it come from multiple IP addresses. There are a few IRC botnets that can do it.